hubploy/__main__.py

```diff
@@ -74,16 +74,17 @@ def main():
             print("Specify --commit-range manually, or pass --check-registry", file=sys.stderr)
             sys.exit(1)
 
-        if args.push or args.check_registry:
-            auth.registry_auth(args.deployment)
+        # Move this line to auth.py so we always use context manager
+        #if args.push or args.check_registry:
+        with auth.registry_auth(args.deployment, args.push, args.check_registry) as registry_auth_code:
```

Comment:
```suggestion
        with auth.registry_auth(args.deployment, args.push, args.check_registry):
```

I *think* you don't need to use the return value of the context manager

Reply:
You are correct!